### PR TITLE
Bump authn-k8s, conjur-api-go, summon and other dependencies

### PR DESCRIPTION
**Pull request by szh**
### Desired Outcome

- Use the latest versions of conjur-authn-k8s-client with support for JWT and tracing.
- Use the latest version of other dependencies.

### Implemented Changes

- Updated versions of dependencies
- Modified `internal/providers/conjur/provider.go` to use the new authn flow that supports JWT
- Added `go:build ignore` flags to sample CRD go files to avoid compilation errors. These files are provided for documentation purposes and are not meant to be compiled with the project.

### Connected Issue/Story

CyberArk internal issue link: [ONYX-16170](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-16170)

### Definition of Done

- [x] Using latest version of cyberark dependencies
- [x] All tests pass

#### Changelog

- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

**Review by tzheleznyak — DISMISSED** [review]
LGTM

**Comment by szh** [comment]
Waiting to update until new authn-k8s-client is released

File: go.mod
Comment (tzheleznyak) [inline go.mod:8]: Do we want to wait for release so here it will be just 0.23.0 ?

Comment (tzheleznyak) [reply go.mod:8]: And in secrets-provider too?

File: internal/providers/conjur/provider.go
Comment (diverdane) [inline internal/providers/conjur/provider.go:98]: Since you're making changes in this section of the code...  This may be a personal preference sort of thing, but instead of using a but instead of bunch of `if`...`if-else`...`if-else` statements, we could use a switch:
```
switch {
case provider.Username != "" && provider.APIKey != "":
    // Conjur provider using API key. Do stuff
case tokenFile != "":
    // Conjur provider using access token. Do stuff
case provider.AuthnURL != "" && strings.Contains(provider.AuthnURL, "authn-k8s"):
    // Conjur provider using authenticator. Do stuff
default:
    // Return error
}
```
For me, the switch is easier to see what's going on, but your call, don't want to hold up on this.

File: CHANGELOG.md
Comment (diverdane) [inline CHANGELOG.md:12]: I'm wondering if we should include a CHANGELOG.md entry to say that we bumped K8s API version to 1.23.
This *might* be considered a user-facing change, since it affects in which Kubernetes clusters the SP can be run.
(Although this may only affect users who are using CRD/CRs to configure the SP... since I think that's why we're depending on Kubernetes API client and machinery?).

File: go.mod
Comment (diverdane) [inline go.mod:25]: Nice work. These K8s updates are always so tricky.

**Review by diverdane — COMMENTED** [review]
Looks great!!! Thanks to you and @tzheleznyak for getting this moving along!
A couple of suggestions, but your call on those.

File: CHANGELOG.md
Comment (szh) [reply CHANGELOG.md:12]: I believe it only affect CRDs

File: internal/providers/conjur/provider.go
Comment (szh) [reply internal/providers/conjur/provider.go:98]: Looking at the Effective Go [style guide](https://go.dev/doc/effective_go#switch):
> It's therefore possible—**and idiomatic**—to write an if-else-if-else chain as a switch
So I'll make this change

Comment (diverdane) [reply internal/providers/conjur/provider.go:102]: Nice, it's much easier to read!

**Review by diverdane — APPROVED** [review]
LGTM!!!

**Pull request merged**
